File: src/main/java/cookiedragon/luchadora/managers/BindManager.java
package cookiedragon.luchadora.managers;

import cookiedragon.luchadora.event.api.EventDispatcher;
import cookiedragon.luchadora.event.api.Subscriber;
import cookiedragon.luchadora.event.client.KeyPressedEvent;
import cookiedragon.luchadora.module.AbstractModule;
import cookiedragon.luchadora.module.ModuleManager;

/**
 * @author cookiedragon234 15/Dec/2019
 */
public class BindManager
{
	public static void init()
	{
		EventDispatcher.register(BindManager.class);
		EventDispatcher.subscribe(BindManager.class);
	}
	
	@Subscriber
	private static void onKeyPress(KeyPressedEvent event)
	{
		for (AbstractModule module : ModuleManager.getModules())
		{
			if (module.getKeyBind().getValue() == event.getKey())
			{
				module.toggle();
			}
		}
	}
}

File: src/main/kotlin/cookiedragon/luchadora/managers/BindManager.kt
package cookiedragon.luchadora.managers

import cookiedragon.eventsystem.EventDispatcher.Companion.register
import cookiedragon.eventsystem.EventDispatcher.Companion.subscribe
import cookiedragon.eventsystem.Subscriber
import cookiedragon.luchadora.event.client.KeyPressedEvent
import cookiedragon.luchadora.util.Initialisable

/**
 * @author cookiedragon234 15/Feb/2020
 */
object BindManager: Initialisable() {
	init {
		register(this)
		subscribe(this)
	}
	
	@Subscriber
	private fun onKeyPress(event: KeyPressedEvent) {
		for (module in ModuleManager.getModules()) {
			if (module.keyBind.value == event.key) {
				module.toggle()
			}
		}
	}
}
